lib/llm/src/block_manager/connector/scheduler.rs

```diff
@@ -431,7 +431,22 @@ impl Scheduler {
             .get_mut(&request.request_id)
             .expect("slot does not exist");
 
-        unimplemented!("@ziqi")
+        if let Some(worker_requests) = self.enqueued_requests.get_mut(&request.request_id) {
```

Comment:
you might consider doing a remove as the first op, since we will need to do a remove regardless.

- remove
- if some, validate it's the other half (the opposite type in the either) and that the uuids match (extra sanity check)
- enqueue to next stage.

Reply:
to be more clear,

perhaps something like

```
    let inner = out.entry(request_id).or_default();
    match inner.remove(&uuid) {
        Some(existing) => {
            // if this was the arrival of the left, check that existing is the right variant of the other
            // opposite for the other
            Some(existing),
        }
        None => {
            // first arrival case
            // if this is the one assigned to the left, create the either and insert
            // other wise do so for the right.
            inner.insert(uuid, new_val);
            None
        }
    }
```